File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../specification/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../specification/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeSpecificationList = function (idList) {
        return $http.get('../specification/removeSpecification.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id=' + id)
    };

    //修改
    this.updateSpecification = function (specification) {
        return $http.post('../specification/updateSpecification.do', specification)
    };

    //新增
    this.saveSpecification = function (specification) {
        return $http.post('../specification/saveSpecification.do', specification)
    };

    //查询所有规格
    this.selectSpecList = function () {
        return $http.post('../specification/selectSpecList.do')
    };
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import java.util.Map;

import com.pinyougou.group.Specification;
import com.pinyougou.pojo.TbSpecification;
import entity.PageResult;
import entity.Result;
/**
 * specification服务层接口
 * @author jieweili
 *
 */
public interface SpecificationService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbSpecification> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param specification
     * @return
     */
    public Result saveSpecification(Specification specification);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    Specification getSpecificationById(Long id);

    /**
     * 修改对象
     * @param specification
     * @return
     */
    Result updateSpecification(Specification specification);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeSpecification(Long[] idList);

    /**
     * 根据条件分页查询
     * @param specification
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listSpecificationByExample(TbSpecification specification, Integer pageNum, Integer pageSize);

    /**
     * 查询所有规格 封装map
     */
    List<Map> selectSpecList();

}
